tests/DependencyInjection/Compiler/AddProvidersPassTest.php

```diff
@@ -84,4 +88,43 @@ public function testProcessForManyProviders()
         $providersPass = new AddProvidersPass();
         $providersPass->process($containerBuilderMock);
     }
+
+    public function testPriorityRegisteredProviders()
+    {
+        $expectedProviders = [new Reference('id2'), new Reference('id')];
+
+        if (class_exists(IteratorArgument::class)) {
+            $expectedProviders = new IteratorArgument($expectedProviders);
+        }
+
+        $definitionMock = $this->createMock('Symfony\Component\DependencyInjection\Definition');
```

Comment:
please use `Definition::class` for this, with a `use` statement for the Definition class. same for the mocking below